library/network_connections.py

```diff
@@ -356,6 +356,19 @@ def ifcfg_create(
                 connection["parent"], connections, idx
             )
             ifcfg["VID"] = str(connection["vlan"]["id"])
+        elif connection["type"] == "wireless":
+            ifcfg["TYPE"] = "Wireless"
+            ifcfg["ESSID"] = str(connection["wireless"]["ssid"])
+            ifcfg["KEY_MGMT"] = str(connection["wireless"]["key-mgmt"]).upper()
+            ifcfg["MAC_ADDRESS_RANDOMIZATION"] = "default"
+
+            # need to set this by default.
```

Comment:
I couldn’t get the link to come up without it when using init scripts. I then figured out that if you don’t use NetworkManager, you need to set up wireless config manually in wpa_supplicant.conf. This is the main reason I am going to do a new PR without wireless as I think it needs more work (to write wpa_supplicant config files) if we want to fully support using initscripts.

Reply:
I think it would be fine if Wi-Fi is only supported by NetworkManager (at least, for the begining). I don't think many people use initscripts to setup Wi-Fi, do they? It's only important to get a clear error message when the user tries Wi-Fi with initscripts.